src/state_manager/mod.rs

```diff
@@ -752,8 +753,27 @@ where
             .await?
     }
 
-    /// Performs a state transition, and returns the state and receipt root of
-    /// the transition.
+    /// Conceptually, a [`Tipset`] consists of _blocks_ which share an _epoch_.
+    /// Each _block_ contains _messages_, which are executed by the _Filecoin Virtual Machine_
+    /// in a _transaction_.
+    ///
+    /// VM transaction execution essentially looks like this:
+    /// ```text
+    /// previous_state * transaction = new_state
+    /// ```
```

Comment:
This picture might be too simplistic. You cannot get the next state if you only have the previous state and a transaction. Transactions may run query states as far back as 1800 epochs. So we need 1800 states and a transaction to compute the next state.